packages/rocketchat-livechat/imports/server/rest/messages.js

```diff
@@ -0,0 +1,71 @@
+import LivechatVisitors from '../../../server/models/LivechatVisitors';
+
+function checkAuthentication(request) {
+	const receivedToken = request.headers['x-rocketchat-livechat-token'];
+	const secretToken = RocketChat.settings.get('Livechat_secret_token');
+	return receivedToken && receivedToken === secretToken;
+}
+
+RocketChat.API.v1.addRoute('livechat/status/:visitorToken', {
+	get() {
+		if (!checkAuthentication(this.request)) {
+			return RocketChat.API.v1.unauthorized();
+		}
+
+		const info = Meteor.call('livechat:getInitialData', this.urlParams.visitorToken);
+		return RocketChat.API.v1.success(info);
+	}
+});
+
+RocketChat.API.v1.addRoute('livechat/messages', {
+	post() {
+		if (!checkAuthentication(this.request)) {
+			return RocketChat.API.v1.unauthorized();
+		}
+
+		if (!this.bodyParams.visitor) {
+			return RocketChat.API.v1.failure('Body param "visitor" is required');
+		}
+		if (!this.bodyParams.visitor.token) {
+			return RocketChat.API.v1.failure('Body param "visitor.token" is required');
+		}
+		const visitorToken = this.bodyParams.visitor.token;
```

Comment:
Should check `this.bodyParams.messages` as an array and if not empty

Reply:
@rodrigok , we added some validations to `messages` param according to suggestions